server/repository/repository.go

```diff
@@ -94,14 +94,15 @@ func (s *Server) List(ctx context.Context, q *repositorypkg.RepoQuery) (*appsv1.
 
 // Get return the requested configured repository by URL and the state of its connections.
 func (s *Server) Get(ctx context.Context, q *repositorypkg.RepoQuery) (*appsv1.Repository, error) {
-	if err := s.enf.EnforceErr(ctx.Value("claims"), rbacpolicy.ResourceRepositories, rbacpolicy.ActionGet, q.Repo); err != nil {
+	repo, err := s.db.GetRepository(ctx, q.Repo)
+	if err != nil {
 		return nil, err
 	}
 
-	repo, err := s.db.GetRepository(ctx, q.Repo)
-	if err != nil {
+	if err := s.enf.EnforceErr(ctx.Value("claims"), rbacpolicy.ResourceRepositories, rbacpolicy.ActionGet, createRBACObject(repo.Project, repo.Repo)); err != nil {
 		return nil, err
 	}
```

Comment:
I guess returning the error of GetRepository before doing the EnforceErr check could possibly result in [unwanted information disclosure](https://cwe.mitre.org/data/definitions/200.html). 

Can we return a permission denied here maybe?

Reply:
Good comment, thank you